test/utils/fixture-helpers.ts

```
@@ -0,0 +1,175 @@
+        console.log(`Account #${i}: ${account.address} (${balance.toLocaleString()} nRLC)`);
+    }
+}
+
+/**
+ * Transfers ownership of all contracts
+ */
+export async function transferAllOwnerships(chainConfig: any) {
+    if (chainConfig.v5) {
+        const registries = [
+            { name: 'AppRegistry', address: chainConfig.v5.AppRegistry },
+            { name: 'DatasetRegistry', address: chainConfig.v5.DatasetRegistry },
+            { name: 'WorkerpoolRegistry', address: chainConfig.v5.WorkerpoolRegistry },
+        ];
+        for (const registry of registries) {
+            if (registry.address) {
+                await transferRegistryOwnership(registry.name, registry.address);
+            }
+        }
+        if (chainConfig.v5.ERC1538Proxy) {
+            await transferProxyOwnership(chainConfig.v5.ERC1538Proxy);
+        }
+    }
+}
+
+/**
+ * Transfers ownership of a registry
+ */
+export async function transferRegistryOwnership(registryName: string, registryAddress: string) {
+    const accounts = await getIexecAccounts();
+    const newIexecAdminAddress = accounts.iexecAdmin.address;
+    try {
+        const registry = Registry__factory.connect(registryAddress, ethers.provider);
+        const currentOwner = await registry.owner();
+        if (currentOwner.toLowerCase() !== newIexecAdminAddress.toLowerCase()) {
+            console.log(
+                `Transferring ${registryName} ownership from ${currentOwner} to iexecAdmin: ${newIexecAdminAddress}`,
+            );
+            const ownerSigner = await ethers.getImpersonatedSigner(currentOwner);
+            await registry
+                .connect(ownerSigner)
+                .transferOwnership(newIexecAdminAddress)
+                .then((tx: any) => tx.wait());
+        } else {
+            console.log(`${registryName} already owned by iexecAdmin`);
+        }
+    } catch (error) {
+        console.error(`Error transferring ownership of ${registryName}:`, error);
+    }
```

It's better to let the Error be raised causing the script to exit. Otherwise we could be debugging non trivial errors.

---

[fix: propagate error in transferRegistryOwnership function](https://github.com/iExecBlockchainComputing/PoCo/pull/216/commits/1698ead8f75e6c7a1a0f6a9d7b496a95faceae0a)